docs/_tooling/extensions/score_metamodel/log.py

```diff
@@ -10,34 +10,47 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 # *******************************************************************************
+import os
+from pathlib import Path
+
+from sphinx.application import Sphinx
 from sphinx_needs.data import NeedsInfoType
 from sphinx_needs.logging import SphinxLoggerAdapter
 
 
 class CheckLogger:
-    def __init__(self, log: SphinxLoggerAdapter):
+    def __init__(self, log: SphinxLoggerAdapter, app: Sphinx):
         self._log = log
         self._count = 0
+        self._app = app
 
     @staticmethod
-    def _location(need: NeedsInfoType):
+    def _location(need: NeedsInfoType, app: Sphinx):
         def get(key):
             return need.get(key, None)
 
         if get("docname") and get("doctype") and get("lineno"):
             # Note: passing the location as a string allows us to use readable relative paths,
             # passing as a tuple results in absolute paths to ~/.cache/.../bazel-out/..
-            return f"{need['docname']}{need['doctype']}:{need['lineno']}"
+            if "RUNFILES_DIR" in os.environ or "RUNFILES_MANIFEST_FILE" in os.environ:
+                matching_file = f"{need['docname']}{need['doctype']}"
+            else:
+                prefix = Path(app.srcdir).relative_to(Path.cwd())
+                matching_file = f"{prefix}/{need['docname']}{need['doctype']}"
```

Comment:
I would prefer this logic outside of CheckLogger, as introducing a dependency to "app" is a huge dependency?!